site/src/components/SideNavigation/SideNavigation.js

```diff
@@ -1,53 +1,112 @@
 import React from 'react';
 import _ from 'lodash';
 import { StyledListItem, StyledLink } from './styles';
+import { findDOMNode } from 'react-dom';
 
-import { Box } from '@sparkpost/matchbox';
+import { ExpandableMenuItem } from '../ExpandableMenuItem';
+import { Box, WindowEvent } from '@sparkpost/matchbox';
+
+function Section(props) {
+  const { value, navItems } = props;
+
+  const section = navItems.filter(({ section }) => section === value);
+
+  const isActiveSection = section.some(item => {
+    return item.selected;
+  });
+
+  return (
+    <ExpandableMenuItem
+      key={value}
+      value={value}
+      defaultExpanded={isActiveSection}
+      firstRoute={section[0].path}
+    >
+      <List items={section} />
+    </ExpandableMenuItem>
+  );
+}
+
+function List(props) {
+  const { items } = props;
+
+  return (
+    <Box as="ul" mb="600" p="0">
+      {items.map(item => (
+        <StyledListItem key={item.path} selected={item.selected}>
+          <StyledLink
+            to={item.path}
+            disabled={item.disabled}
+            selected={item.selected}
+          >
+            {item.label}
+          </StyledLink>
+        </StyledListItem>
+      ))}
+    </Box>
+  );
+}
+
+function getRectFor(node) {
+  const rect = findDOMNode(node);
+
+  if (!rect) {
+    return {};
+  }
+
+  return rect.getBoundingClientRect();
+}
 
 function SideNavigation(props) {
   const { navItems = [] } = props;
+
+  const ref = React.useRef();
+  const [fixed, setFixed] = React.useState(false);
+  const [position, setPosition] = React.useState({});
+
   const onlyActive = navItems.filter(({ disabled }) => !disabled);
   const rootItems = onlyActive.filter(({ section }) => !section);
   const sectionedItems = onlyActive.filter(({ section }) => !!section);
   const sections = _.uniq(
     sectionedItems.map(({ section }) => section || 'rootList')
   );
 
-  function renderList(list) {
-    return (
-      <Box as="ul" mb="600" p="0">
-        {list.map(item => (
-          <StyledListItem key={item.path} selected={item.selected}>
-            <StyledLink
-              to={item.path}
-              disabled={item.disabled}
-              selected={item.selected}
-            >
-              {item.label}
-            </StyledLink>
-          </StyledListItem>
-        ))}
-      </Box>
-    );
+  function renderSection(key) {
+    return <Section key={key} value={key} navItems={navItems} />;
   }
 
-  function renderSection(key) {
-    const section = navItems.filter(({ section }) => section === key);
-    return (
-      <div key={key}>
-        <Box mt="700" mb="300" fontSize="300" fontWeight="500" color="gray.500">
-          {key}
-        </Box>
-        {renderList(section)}
-      </div>
-    );
+  React.useEffect(function() {
+    setPosition(getRectFor(ref.current));
+  }, []);
+
+  function onScroll() {
+    let { scrollY } = window;
+
+    if (scrollY >= position.top) {
+      setFixed(true);
```

Comment:
I think you can do this with `position="sticky"` also instead of using javascript